datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -220,6 +222,21 @@ def create_scraper_configuration(self, instance=None):
         # INTERNAL FEATURE, might be removed in future versions
         config['_text_filter_blacklist'] = instance.get('_text_filter_blacklist', [])
 
+        # Whether or not to use the service account bearer token for authentication
+        # if 'bearer_token_path' is not set, we use /var/run/secrets/kubernetes.io/serviceaccount/token
+        # as a default path to get the token.
+        config['bearer_token_auth'] = is_affirmative(
+            instance.get('bearer_token_auth', default_instance.get('bearer_token_auth', False))
+        )
+
+        # Can be used to get a service account bearer token from files
+        # other than /var/run/secrets/kubernetes.io/serviceaccount/token
+        # 'bearer_token_auth' should be enabled.
+        config['bearer_token_path'] = instance.get('bearer_token_path', default_instance.get('bearer_token_path', None))
+
+        # The service account bearer token to be used for authentication
+        config['_bearer_token'] = self._get_bearer_token(config['bearer_token_auth'], config['bearer_token_path'])
```

Comment:
If I understand correctly the code: in case of the token file is missing, only a log error will inform the user. Then the check will run without "Authentication Header" ?

Do we really want to run the check in case of bad configuration?

Reply:
your understanding of the code is correct, we don't raise the error because it's an optional param so we could try to scrap metrics anyway and if we get authorization error it will be raised by the `poll` function, so the user will see both correlated errors (bad config and http error).

If you think it would be better to raise the exception sooner we can do it in `_get_bearer_token` :)

Reply:
In case of file not found the error return by the server will be misleading; something like: error `HTTP 403 Unauthorised user` instead of a functional error: bearer token file not found.

Reply:
makes sense 👍